frontend/ui/detailview.jsx

```diff
@@ -420,6 +428,36 @@ class DetailViewComponent extends React.Component {
   }
 
   render() {
+    const legends = this.getLegend().sort((a, b) => a.title < b.title);
+    const {
+      measure,
+      platform,
+      timeInterval
+    } = this.state;
+    let crashStatsLink;
+    if (measure in MEASUREMENTS) {
+      const { processType, extraParams } = MEASUREMENTS[measure];
+      const now = Date.now();
+      let queryParams;
+      if (extraParams) {
+        queryParams = `${stringify(Object.keys(extraParams)
+          .reduce((dict, key) => ({ ...dict, [key]: extraParams[key] }), {}))}&`;
+      }
+      queryParams += stringify({
+        product: 'Firefox',
+        version: legends.map(({ title }) => title),
+        platform,
+        process_type: processType,
+        date: [
+          `>=${new Date(new Date(now - (timeInterval * 1000))).toISOString()}`,
+          `<${new Date(now).toISOString()}`
```

Comment:
Please see if you can't use momentjs (http://momentjs.com/) to make this a bit more concise.